modules/auth/db/hash.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package db
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// DefaultHasherStruct stores the available hashing instances
+type DefaultHasherStruct struct {
+	Hashers map[string]Hasher
+}
+
+// Hasher is the interface for a single hash implementation
+type Hasher interface {
+	Validate(password, hash, salt, config string) bool
+	HashPassword(password, salt, config string) (string, string, error)
+	getConfigFromSetting() string
+	getConfigFromAlgo(algo string) string
+}
+
+// HashPassword returns a PasswordHash, PassWordAlgo (and optionally an error)
+func (d DefaultHasherStruct) HashPassword(password, salt, config string) (string, string, error) {
+	if setting.PasswordHashAlgo == "" {
+		setting.PasswordHashAlgo = "pbkdf2"
+	}
```

Comment:
Yes, it was because of the tests. There was a fallback [in the old code](https://github.com/go-gitea/gitea/blob/master/models/user.go#L400), so I thought it's okay. But I'm even more a fan of not having this. So I will modify the tests...